relay-event-normalization/src/normalize/nel.rs

```diff
@@ -1,90 +1,69 @@
+    TraceId,
 };
-use relay_protocol::Annotated;
+use relay_protocol::{Annotated, Object};
 
-/// Enriches the event with new values using the provided [`NetworkReportRaw`].
-pub fn enrich_event(event: &mut Event, nel: Annotated<NetworkReportRaw>) {
-    // If the incoming NEL report is empty or it contains an empty body, just exit.
-    let Some(nel) = nel.into_value() else {
-        return;
-    };
-    let Some(body) = nel.body.into_value() else {
-        return;
-    };
-
-    event.logger = Annotated::from("nel".to_string());
+/// Creates a [`OurLog`] from the provided [`NetworkReportRaw`].
+pub fn create_log(nel: Annotated<NetworkReportRaw>, received_at: DateTime<Utc>) -> Option<OurLog> {
+    let nel = nel.into_value()?;
+    let body = nel.body.into_value()?;
 
-    event.logentry = Annotated::new(LogEntry::from({
-        if nel.ty.value().map_or("<unknown-type>", |v| v.as_str()) == "http.error" {
-            format!(
-                "{} / {} ({})",
-                body.phase.as_str().unwrap_or("<unknown-phase>"),
-                body.ty.as_str().unwrap_or("<unknown-type>"),
-                body.status_code.value().unwrap_or(&0)
-            )
-        } else {
-            format!(
-                "{} / {}",
-                body.phase.as_str().unwrap_or("<unknown-phase>"),
-                body.ty.as_str().unwrap_or("<unknown-type>"),
-            )
-        }
-    }));
+    let message = if nel.ty.value().map_or("<unknown-type>", |v| v.as_str()) == "http.error" {
+        format!(
+            "{} / {} ({})",
+            body.phase.as_str().unwrap_or("<unknown-phase>"),
+            body.ty.as_str().unwrap_or("<unknown-type>"),
+            body.status_code.value().unwrap_or(&0)
+        )
+    } else {
+        format!(
+            "{} / {}",
+            body.phase.as_str().unwrap_or("<unknown-phase>"),
+            body.ty.as_str().unwrap_or("<unknown-type>"),
+        )
+    };
 
-    let request = event.request.get_or_insert_with(Request::default);
-    request.url = nel.url;
-    request.method = body.method;
-    request.protocol = body.protocol;
+    let timestamp = received_at
+        .checked_sub_signed(Duration::milliseconds(*nel.age.value().unwrap_or(&0)))
+        .unwrap_or(received_at);
 
-    let headers = request.headers.get_or_insert_with(Headers::default);
+    let mut attributes: Object<Attribute> = Default::default();
 
-    if let Some(ref user_agent) = nel.user_agent.value() {
-        if !user_agent.is_empty() {
-            headers.insert(
-                HeaderName::new("user-agent"),
-                HeaderValue::new(user_agent).into(),
+    macro_rules! add_attribute {
+        ($name:literal, $value:expr) => {{
+            attributes.insert(
+                $name.to_owned(),
+                Annotated::new(Attribute {
+                    value: AttributeValue {
+                        ty: Annotated::new(AttributeType::String),
```

Comment:
Is this right? In the test in `test_nel.py` this seems to be writing number attributes as strings.